app/questionnaire/questionnaire_manager.py

```diff
@@ -149,9 +149,16 @@ def _get_next_answer_instance(answer_store, answer_id):
         return next_answer_instance_id
 
     def remove_answer(self, location, answer_store, index_to_remove):
-        answer = self.state.get_answers()[int(index_to_remove)]
-        question = answer.parent
-        question.remove_answer(answer)
 
-        answer_store.remove(answer.flatten())
+        state_answers = self.state.get_answers()
+        answers_to_remove = []
+        for state_answer in state_answers:
+            if state_answer.answer_instance == index_to_remove:
+                answers_to_remove.append(state_answer)
+
+        for answer_to_remove in answers_to_remove:
```

Comment:
can this logic not be in the loop above rather than creating a list and instantly looping it?